tests/aws_iot_secure_tunneling_client_test.c

```diff
@@ -135,6 +145,24 @@ int main(int argc, char **argv) {
     ASSERT_SUCCESS(aws_condition_variable_wait(&condition_variable, &mutex));
     aws_mutex_unlock(&mutex);
 
+    if (local_proxy_mode == AWS_SECURE_TUNNELING_SOURCE_MODE) {
+        AWS_RETURN_ERROR_IF2(aws_secure_tunnel_stream_start(secure_tunnel) == AWS_OP_SUCCESS, AWS_OP_ERR);
+        ASSERT_SUCCESS(aws_condition_variable_wait(&condition_variable, &mutex));
```

Comment:
Interesting behavior of secure tunneling. Source side doesn't necessarily have to wait to flush StreamStart(stream_id=N) before following up with StreamData(stream_id=N) as long as the StreamStart is sent first. If a StreamStart succeeds on the destination side, there is no response. If the destination has data to send back, it'll send it back with StreamData on the same stream id as the StreamStart. If it fails, a StreamReset(stream_id=N) will be sent back and the source side will have the necessary state to understand that stream has failed and stop sending any more data. Any data in-flight across the tunnel doesn't matter, and may be dropped by the components in between, including getting to the destination local proxy on the other end and that drops any data messages following the failed StreamStart attempt.

Reply:
I ran the master branch (has Francis's receive data code) on the destination side to make sure data goes all the way through.  I also tested to verify the localproxy could read the data on destination side as well with netcat.  As for the waiting here I was using sleep before switching to the condition var, just so that I could read the logs without them getting blown up before.  I understand we don't necessarily need to wait.